rust/headless-client/src/ipc_service.rs

```diff
@@ -311,8 +311,14 @@ impl<'a> Handler<'a> {
                     break HandlerOk::Err;
                 }
                 Event::Ipc(msg) => {
+                    let msg_variant = serde_variant::to_variant_name(&msg)
+                        .expect("IPC messages should support `to_variant_name`");
                     if let Err(error) = self.handle_ipc_msg(msg).await {
```

Comment:
```suggestion
	            let _entered = tracing::error_span!("handle_ipc_msg", msg = %msg_variant).entered();
                    if let Err(error) = self.handle_ipc_msg(msg).await {
```

if you add this, you don't need to add the field to the error and it will also be present in all messages logged _within_ the function. Could be useful :)